dialogue-apache-hc4-client/src/main/java/com/palantir/dialogue/hc4/ApacheHttpClientChannels.java

```diff
@@ -75,15 +75,26 @@
 
     private ApacheHttpClientChannels() {}
 
-    public static Channel create(ClientConfiguration conf, UserAgent baseAgent) {
+    public static Channel create(ClientConfiguration conf, UserAgent agent) {
+        CloseableHttpClient client = createCloseableHttpClient(conf);
+        List<Channel> channels =
+                conf.uris().stream().map(uri -> createSingleUri(uri, client)).collect(Collectors.toList());
+
+        return Channels.create(channels, agent, conf);
+    }
+
+    public static Channel createSingleUri(String uri, CloseableHttpClient client) {
+        return BlockingChannelAdapter.of(new ApacheHttpClientBlockingChannel(client, url(uri)));
+    }
+
+    public static CloseableHttpClient createCloseableHttpClient(ClientConfiguration conf) {
```

Comment:
A little bit wonky in public api since it's not clear what functionality the returned client provides compared to our channel wrappers. Will we cause bugs if we move functionality from the client into a channel?
At the same time, this should be an implementation dependency of components which simply provide channels, so it's probably fine.

Reply:
so I wanted to just convey the idea of some closeable shared resource - would you prefer if I wrap it in some opaque type so that people can't use it directly?